### PR TITLE
Support for special characters on the URL + support Unity's compressed build

### PR DESCRIPTION
**Pull request by forestrf**
Support for special characters on the URL + support Unity's compressed build

### CODE REVIEW

File: HttpServer.cs [hunk 1]
```diff
@@ -140,6 +144,7 @@ class HttpServer {
 
   private void Process (HttpListenerContext context) {
     string filename = context.Request.Url.AbsolutePath;
+    filename = HttpUtility.UrlDecode(filename);
```
Comment (krsbx) [inline HttpServer.cs:147]: we could make this one into

string filename = HttpUtility.UrlDecode(context.Request.Url.AbsolutePath);

File: HttpServer.cs [hunk 2]
```diff
@@ -154,22 +159,40 @@ class HttpServer {
...
-        context.Response.ContentLength64 = input.Length;
-        context.Response.AddHeader("Date", DateTime.Now.ToString("r"));
-        context.Response.AddHeader("Last-Modified", System.IO.File.GetLastWriteTime(filename).ToString("r"));
+    bool isValidPath;
+    try {
+      isValidPath = new Uri(_rootDirectory).IsBaseOf(new Uri(filename));
+    }
+    catch (Exception) {
+      isValidPath = false;
+    }
```
Comment (krsbx) [inline HttpServer.cs:168]: For this part, I think we could make it into a functions, so we coudl make it to be like this

bool checkIfValid () {
try{
return ...
}catch(err){
return ...
}
}

**Review by krsbx — CHANGES_REQUESTED** [review]
please change the code

File: HttpServer.cs [hunk 3]
```diff
@@ -82,9 +85,16 @@ class HttpServer {
         {".xml", "text/xml"},
         {".xpi", "application/x-xpinstall"},
         {".zip", "application/zip"},
+        {".wasm", "application/wasm"},
```
Comment (krsbx) [inline HttpServer.cs:87]: for this one, I think its better to split the mimeType into another class, so the HttpServer class only going to contain the code for server handling

File: HttpServer.cs (hunk 1 above)
Comment (krsbx) [inline HttpServer.cs:151]: we could make this one into

string filename = HttpUtility.UrlDecode(context.Request.Url.AbsolutePath);

File: HttpServer.cs (hunk 2 above)
Comment (krsbx) [inline HttpServer.cs]: For this part, I think we could make it into a functions, so we coudl make it to be like this

bool checkIfValid () {
try{
return ...
}catch(err){
return ...
}
}

**Comment by forestrf** [comment]
I'm sorry, I didn't want to commit so many changes at once, I thought a pull request was for a specific commit, not for a branch.
Now there are a lot of different changes that aren't listed on the pull title
Let me do those changes, yes

**Comment by forestrf** [comment]
Changes updated, simplified and the path validation is now in a separate function

**Review by krsbx — APPROVED** [review]
LGTM

**Comment by forestrf** [comment]
I wasn't able to see the comments! I don't know why, now I can see them. That gave me problems to fix the problems you pointed out, so I tried to fix them by aking the code as simple as possible and with as few changes as possible.

**Pull request merged**